### DISCUSSION TITLE
how to use "Global Query Parameters"

### DISCUSSION THREAD

**Discussion by casualmatt** [Intevel/nuxt-directus#20] [Q&A] [answered]
How to use the Global Query Parameters, like limit, filter, fields. 

```
            fetchOneDataParams: async function (id: string, collection: string, params: object): Promise<Data> {
                const { getItemById } = useDirectusItems();
                const returnValue = new Data();
                try {
                    returnValue.data = await getItemById({
                        collection: collection,
                        id: id,
                        params:params,
                    });
                } catch (error) {
                    returnValue.error = String(error);
                }
                return returnValue;
            },
```

my params object is defined in a const, but it doesn't seams to work. 
```
const parameters = {
      fields:[
        "id","status","invoices.number", "invoices.id"
      ]
    }
```
_note: i pass the parameters, id and collection to the function and i check if they ware OK with some old console log._ 

any suggestion?

**Comment by casualmatt** [comment, marked as answer]
Also i think that the deep query parameter is not supported. 
https://docs.directus.io/reference/query/#deep
https://github.com/Intevel/nuxt-directus/blob/ca57dbd41e49b46b540e068706f65e21ee2c021f/src/runtime/types/index.d.ts#L42-L50

**Reply by casualmatt** [reply]
ok, yes parameters are only implemented on `getItems `and not on `getItemByID`

https://github.com/Intevel/nuxt-directus/blob/ca57dbd41e49b46b540e068706f65e21ee2c021f/src/runtime/composables/useDirectusItems.ts#L25-L31

https://github.com/Intevel/nuxt-directus/blob/ca57dbd41e49b46b540e068706f65e21ee2c021f/src/runtime/composables/useDirectusItems.ts#L11-L23

**Reply by Intevel** [reply]
Yes that's right. For `getItemById` it wouldn't make sense.

**Reply by casualmatt** [reply]
It actually does, because one of the main features of directus is this graphql like feature to just fetch some data or related data with one request.
Let's say that i have a dashboard and i want to show some data from various tables that are related to a user.
I can do that with the fields query parameters.

**Reply by Intevel** [reply]
I will add the query paramters support to `getItemById`

**Reply by casualmatt** [reply]
Amazing as always 🤠

**Comment by casualmatt** [comment]
Hello  @Intevel,
got stuck on "Deep" Query Parameters. 

**EDIT: Sorry cache error forgot to set the env to not cache in dev environment, now it all works** 

I try with various options:

```ts
const parameters = {
  fields: [
    "*",
    "invoices.number",
    "invoices.id",
    "invoices.object",
    "invoices.status",
    "subscriptions.id",
    "subscriptions.status",
    "subscriptions.metadata_title",
  ],
  deep: '{"invoices":{"_limit":3,"_sort":"-created"},"subscriptions":{"_limit":3}}',
};

```

This first option also has an interesting error shown by VScode, it says that deep doesn't accept a string. 

or deep as an object and not as a string, doesn't work. 


```ts
const parameters = {
  fields: [
    "*",
    "invoices.number",
    "invoices.id",
    "invoices.object",
    "invoices.status",
    "subscriptions.id",
    "subscriptions.status",
    "subscriptions.metadata_title",
  ],
  deep: {
    invoices: { _limit: 3, _sort: "-created" },
    subscriptions: { _limit: 3 },
  },
};
```
